main.tf

```diff
@@ -0,0 +1,189 @@
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    effect = "Allow"
+
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "es:ESHttpGet",
+      "es:ESHttpPut",
+      "es:ESHttpPost",
+      "es:ESHttpHead",
+      "es:ESHttpDelete",
+      "es:Describe*",
+      "es:List*",
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "${var.es_domain_arn}",
+    ]
+  }
+}
+
+# Modules
+#--------------------------------------------------------------
+module "label" {
+  source     = "git::https://github.com/cloudposse/terraform-terraform-label.git?ref=tags/0.1.6"
+  namespace  = "${var.namespace}"
+  name       = "${var.name}"
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  attributes = "${compact(concat(var.attributes, list("elasticsearch", "cleanup")))}"
+  tags       = "${var.tags}"
+  enabled    = "true"
+}
+
+module "artifact" {
+  source      = "git::https://github.com/cloudposse/terraform-external-module-artifact.git?ref=tags/0.1.1"
+  filename    = "lambda.zip"
+  module_name = "terraform-aws-lambda-elasticsearch-cleanup"
+  module_path = "${path.module}"
+}
+
+# Locals
+#--------------------------------------------------------------
+locals {
+  function_name = "${module.label.id}-${module.label.attributes}"
+  enabled       = "${var.enabled == "true" ? true : false }"
+}
+
+# Resources
+#--------------------------------------------------------------
+resource "aws_lambda_function" "default" {
+  count            = "${local.enabled == true ? 1 : 0}"
+  filename         = "${module.artifact.file}"
+  function_name    = "${local.function_name}"
+  description      = "${local.function_name}"
+  timeout          = "${var.timeout}"
+  runtime          = "python${var.python_version}"
+  role             = "${aws_iam_role.default.arn}"
+  handler          = "es-cleanup.lambda_handler"
+  source_code_hash = "${module.artifact.base64sha256}"
+  tags             = "${module.label.tags}"
+
+  environment {
+    variables = {
+      es_endpoint  = "${var.es_endpoint}"
+      index        = "${var.index}"
+      delete_after = "${var.delete_after}"
+      index_format = "${var.index_format}"
+      sns_alert    = "${var.sns_alert}"
+    }
+  }
+
+  vpc_config {
+    subnet_ids         = ["${var.subnet_ids}"]
+    security_group_ids = ["${aws_security_group.default.id}"]
+  }
+}
+
+resource "aws_security_group" "default" {
+  count       = "${local.enabled == true ? 1 : 0}"
+  name        = "${local.function_name}"
+  description = "${local.function_name}"
+  vpc_id      = "${var.subnet_ids}"
+  tags        = "${module.label.tags}"
+}
+
+resource "aws_security_group_rule" "egress_from_lambda_to_es_cluster" {
+  count                    = "${local.enabled == true ? 1 : 0}"
+  description              = "Allow outbound traffic from Lambda Elasticsearch cleanup SG to Elasticsearch SG"
+  type                     = "egress"
+  from_port                = 443
```

Comment:
port 443 only?

Reply:
Lambda speaks to the ES endpoint over HTTPS. I too expected to need DNS open, but this was not the case during testing.